mailboxes/mailboxes-lib.pl

```diff
@@ -1286,5 +1286,81 @@ sub user_read_dbm_file
 return $rv;
 }
 
+# parse_calendar_file(calendar-file)
+# Parses an iCalendar file and returns a list of events
+sub parse_calendar_file
+{
+my (@events, %event);
+eval "use DateTime; use DateTime::TimeZone;";
+return \@events if ($@);
+my ($calendar_file) = @_;
```

Comment:
Move this line up to the start of the sub

Reply:
But the sub might return early? Why would we want to make a variable declaration and list assignment if they might never be used? Performance impact is negligible but still?

I can see that best practices may be involved, but that will be a trade-off from a logic perspective..

Reply:
Best practice in Webmin code is to assign @_ to the formal function parameters in the first line, even if they are never used.

Reply:
Done! Thanks for providing the details!